src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -763,8 +771,22 @@ void SolverInterfaceImpl::setMeshEdges(
     int        size,
     const int *vertices)
 {
+  PRECICE_TRACE(meshID, size);
+  PRECICE_REQUIRE_MESH_MODIFY(meshID);
+  MeshContext &context = _accessor->usedMeshContext(meshID);
+  if (context.meshRequirement != mapping::Mapping::MeshRequirement::FULL) {
+    return;
```

Comment:
Do we want to document this behavior for the user and or developers?

Reply:
This is part of the docs on the website. You mean to document this in doxygen, right?

Instead of silently doing nothing, we could also warn that the adapter dev should use `requiresMeshConnectivityFor()` to prevent this.

Reply:
Added this to the docs

Reply:
> Instead of silently doing nothing, we could also warn that the adapter dev should use requiresMeshConnectivityFor() to prevent this.

"should" is a bit too strict here in my opinion. `requiresMeshConnectivityFor()` is mainly meant to save compute effort in the adapter, not in preCICE.

Reply:
Let's leave it at the doc then.

Reply:
Problem with warning is that if the user calls `setMeshEdge` 1e6 times, then the logs will be very _very_ verbose.
To we would need to implement some kind of bookkeeping, which then needs to be maintained etc.

Reply:
I explained the behaviour in the API documentation.